File: doc/src/scripts/Connecting.java
import java.io.IOException;
import krpc.client.Connection;
import krpc.client.RPCException;
import krpc.client.services.KRPC;

public class Connecting {
    public static void main(String[] args) throws IOException, RPCException {
        Connection connection = Connection.newInstance("Remote example", "my.domain.name", 1000, 1001);
        System.out.println(KRPC.newInstance(connection).getStatus().getVersion());
    }
}

File: doc/src/scripts/Connecting.cpp
#include <iostream>
#include <krpc.hpp>
#include <krpc/services/krpc.hpp>

int main() {
  krpc::Client conn = krpc::connect("Remote example", "my.domain.name", 1000, 1001);
  krpc::services::KRPC krpc(&conn);
  std::cout << krpc.get_status().version() << std::endl;
}
